File: cPlusPlusCode/rotate.cpp
#include<opencv2/imgproc.hpp>
#include<opencv2/highgui.hpp>
#include<iostream>

using namespace std;
using namespace cv;

int main(){
    //Read the input image over which rotation will be applied
    Mat inputImage, roationMatrix, result;
    inputImage = imread("../assets/putin.jpg");

    //define the center using Point2f datatype and rotation angle
    Point2f center(inputImage.cols/2, inputImage.rows/2);
    double rotatingAngle = -30;
    double scale = 1;

    //Get the matrix which defines rotation and will be used to rotate image. getRotationMatrix2D takes center(point around which rotation will be applied),
    //rotation Angle (angle by which image will be rotated), scale:(an optional scaling factor)
    roationMatrix = getRotationMatrix2D(center, rotatingAngle, scale);
    cout<<roationMatrix<<endl;

    //Rotating the input image
    warpAffine(inputImage, result, roationMatrix, Size(inputImage.cols, inputImage.rows));

    //create windows to display images
    namedWindow("image", WINDOW_NORMAL);
    namedWindow("rotated", WINDOW_NORMAL);

    //display images
    imshow("image", inputImage);
    imshow("rotated", result);

    //Press esc to exit the program
    waitKey(0);

    //close all the opened windows
    destroyAllWindows();

    return 0;
}

File: pythonCode/rotate.py
import cv2

inputImage = cv2.imread("../assets/putin.jpg")

center = (inputImage.shape[1]/2, inputImage.shape[0]/2)
rotationAngle = -30
scale = 1

roationMatrix=cv2.getRotationMatrix2D(center, rotationAngle, scale)
size = (inputImage.shape[1], inputImage.shape[0])

result = cv2.warpAffine(inputImage, roationMatrix, size)

cv2.namedWindow("image", cv2.WINDOW_NORMAL)
cv2.namedWindow("result", cv2.WINDOW_NORMAL)

cv2.imshow("image", inputImage)
cv2.imshow("result", result)

cv2.waitKey(0)
cv2.destroyAllWindows()